packages/x-components/src/plugins/devtools/timeline.devtools.ts

```diff
@@ -0,0 +1,102 @@
+import { DevtoolsPluginApi } from '@vue/devtools-api';
+import { XEvent, XEventPayload } from '../../wiring/events.types';
+import { WirePayload } from '../../wiring/wiring.types';
+
+/**
+ * Contains the devtools API.
+ */
+let DevtoolsAPI: DevtoolsPluginApi<void> | undefined;
+
+/**
+ * List of all the configured timeline layers for {@link XEvent}s.
+ */
+const TimelineLayers = [
+  {
+    id: 'x-components-module-registered-events',
+    regex: /^ModuleRegistered$/,
+    label: 'X-Components registered modules'
+  },
+  {
+    id: 'x-components-user-events',
+    regex: /^User/,
+    label: 'X-Components user events'
+  },
+  {
+    id: 'x-components-status-change-events',
+    regex: /RequestChanged$/,
+    label: 'X-Components status change events'
+  },
+  {
+    id: 'x-components-request-events',
+    regex: /Changed$/,
+    label: 'X-Components request events'
+  },
+  {
+    id: 'x-components-miscellanea-events',
+    regex: /.*/,
+    label: 'X-Components miscellanea events'
+  }
+];
+
+/** Set containing the different layer ids. */
+const layerIds = new Set(TimelineLayers.map(layer => layer.id));
+
+/**
+ * Retrieves the timeline layer that an {@link XEvent} belongs to.
+ *
+ * @param event - The {@link XEvent} to retrieve its layer id.
+ * @returns The layer id for the {@link XEvent}.
+ */
+function getTimelineLayer(event: XEvent): string {
+  return TimelineLayers.find(layer => layer.regex.test(event))!.id;
+}
+
+/**
+ * Configures Vue's devtools timeline with new rows for {@link XEvent}s.
+ *
+ * @param api - Vue's devtools API.
+ * @internal
+ */
+export function setupTimelinePlugin(api: DevtoolsPluginApi<void>): void {
+  DevtoolsAPI = api;
+  TimelineLayers.forEach(layer =>
+    api.addTimelineLayer({
+      id: layer.id,
+      label: layer.label,
+      color: 0xff0084
```

Comment:
I'll move to a variable (just a new one for this color, or add a color prop in the `TimelineLayers` array.